core/templates/dev/head/editor/exploration_editor.html

```diff
@@ -366,6 +366,40 @@
       </div>
     </script>
 
+    <script type="text/ng-template" id="modals/saveValidationFail">
+        <div class="modal-header">
+            <h3>Error in saving</h3>
```

Comment:
I would give this a more specific title (specifying that it's a non-strict validation fail) and then ask people to report the error to us, as this isn't supposed to happen ever.